Source/Core/DolphinQt/Settings/GameCubePane.cpp

```diff
@@ -97,31 +97,25 @@ void GameCubePane::CreateWidgets()
   }
 
   // Add slot devices
-  for (const auto& entry : {std::make_pair(tr("<Nothing>"), EXIDeviceType::None),
-                            std::make_pair(tr("Dummy"), EXIDeviceType::Dummy),
-                            std::make_pair(tr("Memory Card"), EXIDeviceType::MemoryCard),
-                            std::make_pair(tr("GCI Folder"), EXIDeviceType::MemoryCardFolder),
-                            std::make_pair(tr("USB Gecko"), EXIDeviceType::Gecko),
-                            std::make_pair(tr("Advance Game Port"), EXIDeviceType::AGP),
-                            std::make_pair(tr("Microphone"), EXIDeviceType::Microphone)})
+  for (const auto device : {EXIDeviceType::None, EXIDeviceType::Dummy, EXIDeviceType::MemoryCard,
+                            EXIDeviceType::MemoryCardFolder, EXIDeviceType::Gecko,
+                            EXIDeviceType::AGP, EXIDeviceType::Microphone})
   {
-    m_slot_combos[0]->addItem(entry.first, static_cast<int>(entry.second));
-    m_slot_combos[1]->addItem(entry.first, static_cast<int>(entry.second));
+    QString name = tr(fmt::format("{:n}", device).c_str());
+    int value = static_cast<int>(device);
+    m_slot_combos[0]->addItem(name, value);
+    m_slot_combos[1]->addItem(name, value);
   }
 
   // Add SP1 devices
-  std::vector<std::pair<QString, EXIDeviceType>> sp1Entries{
-      std::make_pair(tr("<Nothing>"), EXIDeviceType::None),
-      std::make_pair(tr("Dummy"), EXIDeviceType::Dummy),
-      std::make_pair(tr("Broadband Adapter (TAP)"), EXIDeviceType::Ethernet),
-      std::make_pair(tr("Broadband Adapter (XLink Kai)"), EXIDeviceType::EthernetXLink)};
+  std::vector<EXIDeviceType> sp1Entries{EXIDeviceType::None, EXIDeviceType::Dummy,
+                                        EXIDeviceType::Ethernet, EXIDeviceType::EthernetXLink};
 #if defined(__APPLE__)
-  sp1Entries.emplace_back(std::make_pair(tr("Broadband Adapter (tapserver)"),
-                                         ExpansionInterface::EXIDeviceType::EthernetTapServer));
+  sp1Entries.emplace_back(EXIDeviceType::EthernetTapServer);
```

Comment:
This could be instead:

```
  constexpr auto sp1Entries = {
    EXIDeviceType::None,
    EXIDeviceType::Dummy,
    EXIDeviceType::Ethernet,
    EXIDeviceType::EthernetXLink,
#if defined(__APPLE__)
    EXIDeviceType::EthernetTapServer,
#endif
  };
```

Reply:
Unfortunately that doesn't compile (because `initializer_list` isn't constexpr?), but I was able to write this:

```C++
  static constexpr std::array sp1_entries{
      EXIDeviceType::None,
      EXIDeviceType::Dummy,
      EXIDeviceType::Ethernet,
      EXIDeviceType::EthernetXLink,
#ifdef __APPLE__
      EXIDeviceType::EthernetTapServer,
#endif
  };
```

Reply:
Turns out the freebsd builder doesn't like `std::array` on its own; I've changed this to just putting the list directly in the for loop like was done with the slot a/b devices.